UnicornRender/video/include/unicorn/video/vulkan/VkMesh.hpp

```diff
@@ -78,26 +78,11 @@ class VkMesh
     /** @brief Returns model matrix */
     const glm::mat4& GetModelMatrix() const;
 
-    /** @brief Returns vertices count */
-    uint32_t VerticesSize() const;
+    /** @brief Returns constant reference to unicorn::Mesh */
+    Mesh const& GetMesh() const;
 
-    /** @brief Returns indices count */
-    uint32_t IndicesSize() const;
-
-    /** @brief Returns @c true if mesh is colored and @c false otherwise */
-    bool IsColored() const;
-
-    /** @brief Returns @c true if mesh is in wired mode and @c false otherwise */
-    bool IsWired() const;
-
-    /** @brief Returns true if mesh is visible, false otherwise */
-    bool IsVisible() const;
-
-    /**
-     * @brief Returns color of mesh in RGB format
-     * @return color of mesh in RGB format
-     */
-    glm::vec3 GetColor() const;
+    /** @brief Returns constant shared pointer to unicorn::Material */
+    std::shared_ptr<Material> const GetMaterial() const;
```

Comment:
Not sure if this proxy is needed since you already have `GetMesh()` that can provide all mesh-related data. Would prefer not to proxy without a reason.